packages/core/src/account/smartContractAccount.ts

```diff
@@ -179,24 +202,61 @@ export async function toSmartContractAccount<
   getDummySignature,
   signUserOperationHash,
   encodeUpgradeToAndCall,
-}: ToSmartContractAccountParams<Name, TTransport, TChain>): Promise<
-  SmartContractAccount<Name>
-> {
-  const client = createBundlerClient({
+}: ToSmartContractAccountParams<
+  TEntryPointVersion,
+  Name,
+  TTransport,
+  TChain
+>): Promise<AccountEntryPointRegistry<Name>[TEntryPointVersion]>;
+
+export async function toSmartContractAccount<
+  TEntryPointDef extends EntryPointDefRegistry[TEntryPointVersion],
+  TEntryPointVersion extends EntryPointVersion = TEntryPointDef extends EntryPointDef<
+    infer U
+  >
+    ? U
+    : never,
+  Name extends string = string,
+  TTransport extends Transport = Transport,
+  TChain extends Chain = Chain
+>({
+  transport,
+  chain,
+  entryPoint,
+  source,
+  accountAddress,
+  getAccountInitCode,
+  signMessage,
+  signTypedData,
+  encodeBatchExecute,
+  encodeExecute,
+  getDummySignature,
+  signUserOperationHash,
+  encodeUpgradeToAndCall,
+}: ToSmartContractAccountParams<
+  TEntryPointVersion,
+  Name,
+  TTransport,
+  TChain
+>): Promise<AccountEntryPointRegistry<Name>[EntryPointVersion]> {
+  const _entryPoint: EntryPointDef<EntryPointVersion> =
+    entryPoint ?? getEntryPoint(chain);
```

Comment:
I think we should make `entryPoint` a required parameter instead so that typescript infers the output